### ISSUE TITLE
Upstream the git backend

### ISSUE THREAD

**Issue by bpeetz** [stride-tasks/stride#43]
There is an open issue in TaskChampion regarding Git backend support (https://github.com/GothenburgBitFactory/taskchampion/issues/441). Upstreaming stride's git backend code could enable us to transition fully to TaskChampion's task type internally, thus streamlining the backend switching process by eliminating the need for manual task export and import.

However, considering the planned introduction of plugin-based backends, upstreaming this might not be the most efficient use of resources and time. Furthermore, I'm not so sure about the full features of the git backend and if upstream would want to support them in their current form.

**Comment by djmitche** [comment]
FWIW, the Git backend will, like other sync backends, store encrypted "version" binary blobs and not any kind of intelligible task data. It's not really a "good" use of Git -- the suggestion just makes sense since lots of people know how to host Git repositories for free.

**Comment by bpeetz** [comment]
To my knowledge the current git handling in stride is doing exactly that (i.e., storing encrypted blobs in git).

And to my understanding, the git support is there simply because free (both in terms of libre and gratis) git hosts are there aplenty and using them is easier than renting and setting up a server with something like `taskchampion-sync-server`.
Furthermore, some people are probably still reluctant to try to set up the taskchampion sync server, simply because they have experienced the difficulty of setting up `taskserver` (which is a baseless fear, as the taskchampion server is, based on my experience, not difficult to set up).

@HalidOdat I have seen, that you have started implementing plugins. You mentioned in https://github.com/stride-tasks/stride/issues/4#issuecomment-2377705643, that you plan to support different sync backends via these plugins; Is it even worthwhile to try to upstream the git backend?

**Comment by HalidOdat** [comment]
> To my knowledge the current git handling in stride is doing exactly that (i.e., storing encrypted blobs in git).

Not quite, they are encrypted and encoded in text form, so git can diff and resolve conflicts easily, but I planned to make event more efficient to store them in git.

> Is it even worthwhile to try to upstream the git backend?

It’s probably not worth upstreaming the Git backend, as it could limit flexibility and tie us to the upstream’s constraints. Keeping it independent offers more control.
